src/app/pages/map/map.component.ts

```diff
@@ -0,0 +1,109 @@
+import { Component, OnInit, ViewChild, ElementRef } from "@angular/core";
+import { map, Map, tileLayer, LatLng, marker, latLng, Icon, icon, polyline, point, PolylineOptions, LatLngExpression } from "leaflet";
+import { GamePlayManagerService } from "src/app/services";
+import { Segment, Location as MapLocation, BusColor } from "src/app/types";
+
+@Component({
+  selector: "app-map",
+	templateUrl: "./map.component.html",
+	styleUrls: ["./map.component.scss"],
+})
+export class MapComponent implements OnInit {
+  private _mapController: Map;
+  @ViewChild("map")
+	_mapEl: ElementRef;
+  
+	constructor(
+    private gamePlayManager: GamePlayManagerService,) {
+      gamePlayManager.locationSubject.subscribe({
+        next: (locations) => this._renderLocations(locations)
+      })
+      gamePlayManager.segmentSubject.subscribe({
+        next: (segments) => this._renderSegments(segments)
+      })
+    }
+  
+	ngOnInit() {
+  }
+  
+  ngAfterViewInit() {
+    this._initMap();
+  }
+  
+  private _claimSegment(segment: Segment) {
+    this.gamePlayManager.claimSegment(segment);
+  }
+
+  private _initMap(): void {
+    const hbllLocation: LatLng = new LatLng(40.248157, -111.649150);
```

Comment:
Can’t tell if this was just for testing. But everything looks good

Reply:
I thought that it would be a good location to center the map on initially. It is also where we have class and therefore fakes some location data.